textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcherTest.java

```diff
@@ -0,0 +1,402 @@
+
+        Analyzer analyzer = new StandardAnalyzer();
+        IPredicate predicate = new KeywordPredicate(query, attributeList, analyzer);
+        QueryParser queryParser;
+        if(!buildMultiQueryOnAttributeList) {
+            queryParser = new QueryParser(TestConstants.ATTRIBUTES_PEOPLE[0].getFieldName(), analyzer);
+            queryObj = queryParser.parse(LuceneConstants.SCAN_QUERY);
+        }
+        else {
+            queryObj = createQueryObject(query, attributeList);
+        }
+        IDataReader dataReader = new LuceneDataReader(dataStore, queryObj);
+        indexSearchSourceOperator = new IndexSearchSourceOperator(dataReader);
+        keywordMatcher = new KeywordMatcher(predicate, indexSearchSourceOperator);
+        keywordMatcher.open();
+        List<ITuple> results = new ArrayList<>();
+        ITuple nextTuple = null;
+        while ((nextTuple = keywordMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+        indexSearchSourceOperator.close();
+        return results;
+    }
+
+    /**
+     * For a given string query it gets a list of results
+     *
+     * @param query
+     * @return List<ITuple>
+     * @throws DataFlowException
+     * @throws ParseException
+     */
+    public List<ITuple> getQueryResults(String query) throws DataFlowException, ParseException {
+        String defaultField = TestConstants.ATTRIBUTES_PEOPLE[5].getFieldName();
+
+        ArrayList<Attribute> attributeList = new ArrayList<Attribute>();
+        attributeList.add(TestConstants.ATTRIBUTES_PEOPLE[5]);
+        Analyzer analyzer = new StandardAnalyzer();
+        IPredicate predicate = new KeywordPredicate(query, attributeList, analyzer);
+        QueryParser queryParser = new QueryParser(defaultField, analyzer);
+        queryObj = queryParser.parse(query);
+        IDataReader dataReader = new LuceneDataReader(dataStore, queryObj);
+        indexSearchSourceOperator = new IndexSearchSourceOperator(dataReader);
+        keywordMatcher = new KeywordMatcher(predicate, indexSearchSourceOperator);
+        keywordMatcher.open();
+
+        List<ITuple> results = new ArrayList<ITuple>();
+        ITuple nextTuple = null;
+        while ((nextTuple = keywordMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+        indexSearchSourceOperator.close();
+        return results;
+    }
+
+    /**
+     * Verifies Keyword Matcher on multiword string
+     * @throws Exception
+     */
+    @Test
+    public void testKeywordMatcher() throws Exception {
+        String query = "short tall";
+        List<ITuple> results = getQueryResults(query);
+        List<ITuple> tuples = TestConstants.getSamplePeopleTuples();
+        for(ITuple t : results){
+            boolean contains = TestUtils.contains(tuples, t, Arrays.asList(TestConstants.ATTRIBUTES_PEOPLE));
```

Comment:
In this case you're testing if each ITuple in the "results" is present in "tuples" or not. Why don't you use containsAllResults() and send them both as parameters? 

This will fail only if "results" doesn't contain all the elements present in "tuples". If this is not the case replace this with containsAllResults() to maintain uniformity. 

And "tuples" name is confusing. Could you name it according to the context like "expectedTuples". 

Thanks.